src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -693,6 +715,148 @@ bool GenericConfigurationManagerImpl<ImplClass>::_IsPairedToAccount()
+
+        hash.AddData((uint8_t *)inputBuf, kLenFieldLen + kDeviceIdLen);
+    }
+
+    // Hash the device certificate
+    {
+        size_t certLen;
+
+        // Determine the length of the device certificate.
+        err = Impl()->_GetDeviceCertificate((uint8_t *)NULL, 0, certLen);
+        SuccessOrExit(err);
+
+        // Create a temporary buffer to hold the certificate.  (This will also be used for
+        // the private key).
+        dataBufSize = certLen;
+        dataBuf = (uint8_t *)Platform::Security::MemoryAlloc(dataBufSize);
+        VerifyOrExit(dataBuf != NULL, err = WEAVE_ERROR_NO_MEMORY);
+
+        // Read the certificate.
+        err = Impl()->_GetDeviceCertificate(dataBuf, certLen, certLen);
+        SuccessOrExit(err);
+
+        // Hash the length and value of the device certificate in base-64 form.
+        HashLengthAndBase64Value(hash, dataBuf, (uint16_t)certLen);
+    }
+
+    // Hash the device private key
+    {
+        size_t keyLen;
+
+        // Determine the length of the device private key.
+        err = Impl()->_GetDevicePrivateKey((uint8_t *)NULL, 0, keyLen);
+        SuccessOrExit(err);
+
+        // Read the private key.  (Note that we presume the buffer allocated to hold the certificate
+        // is big enough to hold the private key.  _GetDevicePrivateKey() will return an error in the
+        // unlikely event that this is not the case.)
+        err = Impl()->_GetDevicePrivateKey(dataBuf, dataBufSize, keyLen);
+        SuccessOrExit(err);
+
+        // Hash the length and value of the private key in base-64 form.
+        HashLengthAndBase64Value(hash, dataBuf, (uint16_t)keyLen);
+    }
+
+    // Hash the device pairing code.  If the device does not have a pairing code, hash a zero-length value.
+    {
+        char pairingCode[ConfigurationManager::kMaxPairingCodeLength + 1]; // +1 for terminator
+        char lenStr[kLenFieldLen + 1]; // +1 for terminator
+        size_t pairingCodeLen;
+
+        err = Impl()->_GetPairingCode(pairingCode, sizeof(pairingCode), pairingCodeLen);
+        if (err == WEAVE_DEVICE_ERROR_CONFIG_NOT_FOUND)
+        {
+            pairingCodeLen = 0;
+            err = WEAVE_NO_ERROR;
+        }
+        SuccessOrExit(err);
+
+        snprintf(lenStr, sizeof(lenStr), "%04" PRIX16, (uint16_t)pairingCodeLen);
+
+        hash.AddData((uint8_t *)lenStr, kLenFieldLen);
+        hash.AddData((uint8_t *)pairingCode, pairingCodeLen);
+    }
+
+    hash.Finish(hashBuf);
+
+exit:
+    if (dataBuf != NULL)
+    {
+        Crypto::ClearSecretData(dataBuf, dataBufSize);
+        Platform::Security::MemoryFree(dataBuf);
+    }
+    return err;
+}
+
+template<class ImplClass>
+void GenericConfigurationManagerImpl<ImplClass>::HashLengthAndBase64Value(Platform::Security::SHA256 & hash, const uint8_t * val, uint16_t valLen)
+{
+    constexpr uint16_t kInputBufSize = 80;
+    static_assert(kInputBufSize > 0 && kInputBufSize % 4 == 0, "kInputBufSize must be a positive multiple of 4");
```

Comment:
For portability, I'd recommend using the nlSTATIC_ASSERT macros in nlassert.

Reply:
I don't believe there's any portability issue here as static_assert in this form is C++11 and C++11 is the minimum language level supported by the Device Layer.